### ISSUE TITLE
调用WebPEncoder.fromGif实现gif转webp，webp出现闪的问题

### ISSUE THREAD

**Issue by zhi936689039** [penfeizhou/APNG4Android#191]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/penfeizhou/APNG4Android/issues) and found none

### Issue Info

| Info                         | Value                                                 |      |
| ---------------------------- | ----------------------------------------------------- | ---- |
| Device Info                  | e.g. Huawei Mate 30 pro                               |      |
| System Version               | e.g.  6.0                                             |      |
| APNG4Android Library Version | e.g.  1.0.0                                           |      |
| Repro rate                   | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo project  | e.g. does it happen with our demo project?            |      |
| Demo project link            | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by zhi936689039** [comment]
我这块的代码是

**Comment by zhi936689039** [comment]
我这块的代码是：

**Comment by zhi936689039** [comment]
我这块的代码是：     var gifStreamLoader = FileLoader(file?.absolutePath)
                    var webPEncoder = WebPEncoder.fromGif(gifStreamLoader)

                    //
                    ThreadUtils.getSinglePool().execute{
                        var fos = BufferedOutputStream(FileOutputStream(GifDownloadSaveUtil.getFileName(this@MainActivity,"test.webp")))
                        var bytes = webPEncoder.build()
                        fos.write(bytes)
                        fos.close()
                    }

**Comment by zhi936689039** [comment]
这是https://res.naadi.microparty.com/user/1688545363899.gif gif的地址

**Comment by jingpeng** [comment]
可以参考EncoderTestActivity这里的源码，链接如下https://github.com/penfeizhou/APNG4Android/blob/master/app/src/main/java/com/github/penfeizhou/animation/demo/EncoderTestActivity.java

**Comment by zhi936689039** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by jingpeng** [comment]
我在此基础上稍作修改使用了你的gif：
`new Thread(new Runnable() {
            @Override
            public void run() {

                final byte[] ret = WebPEncoder.fromDecoder(
                        GifDrawable.fromAsset(EncoderTestActivity.this,
                                "1688545363899.gif").getFrameSeqDecoder()).build();
                imageView.post(new Runnable() {
                    @Override
                    public void run() {
                        Glide.with(imageView)
                                .load(ret)
                                .into(imageView);
                    }
                });
            }
        }).start();`

**Comment by jingpeng** [comment]
https://github.com/penfeizhou/APNG4Android/assets/5223226/fcafe28d-2359-431e-95dd-0fbf927e965f

**Comment by jingpeng** [comment]
看起来好像不闪？

**Comment by zhi936689039** [comment]
。果然是我调用的方法有问题，那如果是网络那种gif怎么转的

**Comment by jingpeng** [comment]
建议你先进行本地下载缓存，后以本地路径进行转换操作

**Comment by zhi936689039** [comment]
好，我试试

**Comment by zhi936689039** [comment]
那这个也可以通过png转webp嘛

**Comment by zhi936689039** [comment]
--!不行，兄弟，       ThreadUtils.getSinglePool().execute{
                        val ret = WebPEncoder.fromDecoder(
                            APNGDrawable.fromFile(file?.absolutePath).frameSeqDecoder
                        ).build()
                        ThreadUtils.runOnUiThread{
//                            var fos = BufferedOutputStream(FileOutputStream(GifDownloadSaveUtil.getFileName(this@MainActivity,"test.webp")))
//                            fos.write(ret)
//                            fos.close()
                            iv_pic?.let { Glide.with(this@MainActivity).load(ret).into(it) }
                        }
                    }

**Comment by zhi936689039** [comment]
emm..好像我的用法有问题

**Comment by zhi936689039** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
